src/components/UnstyledLink/UnstyledLink.tsx

```diff
@@ -17,7 +17,7 @@ export const UnstyledLink = memo(
   forwardRef<unknown, UnstyledLinkProps>(function UnstyledLink(props, _ref) {
     const LinkComponent = useLink();
     if (LinkComponent) {
-      return <LinkComponent {...unstyled.props} {...props} />;
+      return <LinkComponent {...unstyled.props} {...props} href={props.url} />;
```

Comment:
@BPScott does this break links? I'm not sure what changed on our end in the last couple of weeks but after some debugging this morning I found that the default skip to content link on frame was being marked as not safe in web [here](https://github.com/Shopify/web/blob/master/packages/@shopify/safe-redirect/index.ts#L66-L68). We can Tuple if you need more context

Reply:
Yo yeah, I think this has the potential for weirdness. It might not *break* react-router / next-router links at runtime but it will cause the typing to get weird as LinkComponent expects the url to be in the `url` prop but now you're pumping it into the href too which is confusing.

Was just chatting with @dfmcphee about this. I not looked at this deeply but I *thiiiink* the better fix for this to identify where we're passing "href" into UnstyledLink, and change it so that it gets passed as the URL instead. Dom mentioned UnstyledButton might be doing something unexpected. Commenting out https://github.com/Shopify/polaris-react/blob/master/src/components/UnstyledButton/UnstyledButton.tsx#L12 might give some insight into where we're passing arbitrary stuff. Having that escape hatch of "any prop is fine here" feels like a problem to me. 

Alternativly is there something silly in web's link component where it is render

Reply:
Thanks Ben! I put up this alternative PR in the meantime which is much safer: https://github.com/Shopify/polaris-react/pull/3599. Going to close this one in a bit unless anyone has any other ideas

Reply:
Part of me wonders if some of the stuff in Web's link component stuff isn't realising anchor links  "#whatever" should be treated as a plain a tag and that url needs to be converted to href.